examples/bearriver/src/FRP/BearRiver.hs

```diff
@@ -176,20 +176,54 @@ after q x = feedback q go
                   ct = if t' < 0 then constant (NoEvent, t') else go
               return ((e, t'), ct)
 
-occasionally :: MonadRandom m 
-             => Time -- ^ The time /q/ after which the event should be produced on average
-             -> b    -- ^ Value to produce at time of event
+occasionally :: (Monad m, RandomGen g) 
+             => g 
+             -> Time 
+             -> b 
              -> SF m a (Event b)
-occasionally tAvg b
+occasionally g tAvg b 
```

Comment:
I think you should be able to write something like:
```haskell
occasionally g tAvg b = runRandS (occasionally tAvg b) g >>> arr fst
```
If this works, I suggest rewriting the other primitives in the same way.